### PR TITLE
Update FastHierarchy.java

### PR DESCRIPTION
**Pull request by pavanupb**

### CODE REVIEW

File: src/main/java/soot/FastHierarchy.java [hunk 1]
```diff
@@ -53,6 +55,8 @@
  */
 public class FastHierarchy {
   private static final Logger LOGGER = LoggerFactory.getLogger(FastHierarchy.class);
+  private final LargeNumberedMap<Type, SmallNumberedMap<SootMethod>> typeToVtbl =
```
Comment (mbenz89) [inline src/main/java/soot/FastHierarchy.java]: Please change to Map<SootClass, Map<String,SootMethod>> and use HashMaps as implementation

File: src/main/java/soot/FastHierarchy.java [hunk 2]
```diff
@@ -763,10 +770,21 @@ private SootMethod resolveMethod(
       Set<SootClass> ignoreList) {
     SootClass concreteType = baseType;
 
+    
+    SootMethod candidate = null;
```
Comment (mbenz89) [inline src/main/java/soot/FastHierarchy.java]: Generate a signature with soot.SootMethod#getSubSignature(java.lang.String, java.util.List<soot.Type>, soot.Type) as key for the inner map of vtable

Comment (mbenz89) [reply src/main/java/soot/FastHierarchy.java]: Use the declaring class as key for the outer map

File: src/main/java/soot/FastHierarchy.java [hunk 3]
```diff
@@ -763,10 +770,21 @@ private SootMethod resolveMethod(
...
+    		return candidate;
+    	}
+    }
+    
     // When there is no proper dispatch found, we simply return null to let
     // the caller decide what to do
-    SootMethod candidate = null;
-
     while (concreteType != null && !ignoreList.contains(concreteType)) {
       ignoreList.add(concreteType);
```
Comment (mbenz89) [inline src/main/java/soot/FastHierarchy.java]: Dont forgett to add the resolved method to the vtable if one was found (at the end of the method)

**Comment by pavanupb** [comment]
@mbenz89 I have changed the vtbl. Please have a look and let me know for any changes

**Pull request merged**